app/main.py

```diff
@@ -1,3 +1,56 @@
 class Distance:
-    # Write your code here
-    pass
+    def __init__(self, km: float) -> None:
+        self.km = km
+
+    def __str__(self) -> str:
+        return f"Distance: {self.km} kilometers."
+
+    def __repr__(self) -> str:
+        return f"Distance(km={self.km})"
+
+    def __add__(self, other: "Distance") -> "Distance":
+        if isinstance(other, Distance):
+            return Distance(self.km + other.km)
+        elif isinstance(other, (int,float)):
+            return Distance(self.km + other)
+        else:
+            return NotImplemented
+
+    def __iadd__(self, other: "Distance") -> "Distance":
+        if isinstance(other, Distance):
+            self.km += other.km
+        elif isinstance(other, (int, float)):
+            self.km += other
+        else:
+            return NotImplemented
+        return self
+
+    def __mul__(self, factor: "Distance") -> "Distance":
+        if isinstance(factor, (int, float)):
+            return Distance(self.km * factor)
+        else:
+            return NotImplemented
+
+    def __truediv__(self, divisor: "Distance") -> "Distance":
```

Comment:
The type hint for the `divisor` parameter in the `__truediv__` method should be `(int, float)` instead of `"Distance"`, as this method is designed to divide the distance by a numeric divisor.